### PR TITLE
Update UploadBehavior.php

### PR DESCRIPTION
**Pull request by EOM**
Fix PHP 5.4 change valid for PHP 5.3

### CODE REVIEW

**Comment by EOM** [comment]
Small changes in the code. 
And then you should use PHP5 (int), not intval () that PHP4

File: Model/Behavior/UploadBehavior.php [hunk 1]
```diff
@@ -1508,12 +1508,12 @@ protected function _createImageResource($filename, $pathInfo) {
...
 	protected function _imagecreatefromjpegexif($filename) {
 		$image = imagecreatefromjpeg($filename);
-		$exif = false;
+		$exif = array();
 		if (function_exists('exif_read_data')) {
 			$exif = exif_read_data($filename);
 		}
 
-		if ($image && $exif && isset($exif['Orientation'])) {
+		if ($image && isset($exif['Orientation']) == true) {
```
Comment (josegonzalez) [inline Model/Behavior/UploadBehavior.php]: `exif_read_data` can return `false`, so assuming it is an array isn't a great idea.

Comment (EOM) [inline Model/Behavior/UploadBehavior.php]: OK

File: Model/Behavior/UploadBehavior.php [hunk 2]
```diff
@@ -1344,7 +1344,7 @@ protected function _resizePhp(Model $model, $field, $path, $size, $geometry, $th
 				$outputHandler = 'imagepng';
 				$supportsQuality = true;
 				// convert 0 (lowest) - 100 (highest) thumbnailQuality, to 0 (highest) - 9 (lowest) quality (see http://php.net/manual/en/function.imagepng.php)
-				$adjustedQuality = intval((100 - $this->settings[$model->alias][$field]['thumbnailQuality']) / 100 * 9);
+				$adjustedQuality = (int)((100 - $this->settings[$model->alias][$field]['thumbnailQuality']) / 100 * 9);
```
Comment (EOM) [inline Model/Behavior/UploadBehavior.php]: This change is OK for Merge, intval() old function PHP4 for format correct cast val in PHP 5. 
```php
$adjustedQuality = (int)((100 - $this->settings[$model->alias][$field]['thumbnailQuality']) / 100 * 9);
```

File: Model/Behavior/UploadBehavior.php (hunk 1 above)
Comment (josegonzalez) [inline Model/Behavior/UploadBehavior.php]: If you revert this, I can merge.

File: Model/Behavior/UploadBehavior.php (hunk 2 above)
Comment (josegonzalez) [inline Model/Behavior/UploadBehavior.php]: Just a note: intval should work just fine in PHP 5.

Comment (EOM) [inline Model/Behavior/UploadBehavior.php]: Hi,
If it works well intval() in PHP 5, but is not recommended to use more it is a function deprecator. Also in the same code is being used cast (int) and for better code quality must remain consistent.

Example: 
UploadBehavior.php 
Lines 508, 605.630, etc ...

**Pull request merged**